sled-agent/src/sled_agent.rs

```diff
@@ -346,21 +346,33 @@ impl SledAgent {
             storage.zone_bundler().clone(),
         )?;
 
-        match config.vmm_reservoir_percentage {
-            Some(sz) if sz > 0 && sz < 100 => {
-                instances.set_reservoir_size(&hardware, sz).map_err(|e| {
-                    error!(log, "Failed to set VMM reservoir size: {e}");
-                    e
-                })?;
-            }
-            Some(sz) if sz == 0 => {
-                warn!(log, "Not using VMM reservoir (size 0 bytes requested)");
-            }
-            None => {
-                warn!(log, "Not using VMM reservoir");
+        // Configure the VMM reservoir as either a percentage of DRAM or as an
+        // exact size in MiB.
+        let reservoir_mode = match (
+            config.vmm_reservoir_percentage,
+            config.vmm_reservoir_size_mb,
+        ) {
+            (None, None) => ReservoirMode::None,
+            (Some(p), None) => ReservoirMode::Percentage(p),
+            (None, Some(mb)) => ReservoirMode::Size(mb),
+            (Some(_), Some(_)) => panic!(
+                "cannot specify vmm_reservoir_percentage and \
+                vmm_reservoir_size_mb at the same time"
```

Comment:
nit: I would probably phrase this as: "only one of `vmm_reservoir_percentage` and `vmm_reservoir_size_mb` is allowed" or something similar (instead of referring to the "same time")

Reply:
Fixed in [23a25d6](https://github.com/oxidecomputer/omicron/pull/4137/commits/23a25d6071e32d2cf4b14cb17cd744d3f30aa814)